caravel/assets/javascripts/SqlLab/components/TabbedSqlEditors.jsx

```diff
@@ -51,6 +53,19 @@ class TabbedSqlEditors extends React.PureComponent {
       window.history.replaceState({}, document.title, this.state.cleanUri);
     }
   }
+  componentWillReceiveProps(nextProps) {
+    const activeQeId = this.props.tabHistory[this.props.tabHistory.length - 1];
+    const newActiveQeId = nextProps.tabHistory[nextProps.tabHistory.length - 1];
+    if (activeQeId !== newActiveQeId || !areObjectsEqual(this.props.queries, nextProps.queries)) {
+      const queriesArray = [];
+      for (const id in this.props.queries) {
+        if (this.props.queries[id].sqlEditorId === newActiveQeId) {
+          queriesArray.push(this.props.queries[id]);
+        }
+      }
+      this.setState({ queriesArray });
```

Comment:
I wonder if it would be kosher to mutate `nextProps` in `componentWillReceiveProps` the docs don't seem to say whether it is...